src/main/scala/firrtl/ExecutionOptionsManager.scala

```diff
@@ -0,0 +1,324 @@
+      case "append" => AppendInfo(inputFileNameOverride)
+      case other => UseInfo
+    }
+  }
+
+  def compiler: Compiler = {
+    compilerName match {
+      case "high"      => new HighFirrtlCompiler()
+      case "low"       => new LowFirrtlCompiler()
+      case "verilog"   => new VerilogCompiler()
+    }
+  }
+
+  def outputSuffix: String = {
+    compilerName match {
+      case "verilog"   => "v"
+      case "low"       => "lo.fir"
+      case "high"      => "hi.fir"
+      case _ =>
+        throw new Exception(s"Illegal compiler name $compilerName")
+    }
+  }
+
+  /**
+    * build the input file name, taking overriding parameters
+    *
+    * @param optionsManager this is needed to access build function and its common options
+    * @return a properly constructed input file name
+    */
+  def getInputFileName(optionsManager: ExecutionOptionsManager): String = {
+    optionsManager.getBuildFileName("fir", inputFileNameOverride)
+  }
+  /**
+    * build the output file name, taking overriding parameters
+    *
+    * @param optionsManager this is needed to access build function and its common options
+    * @return
+    */
+  def getOutputFileName(optionsManager: ExecutionOptionsManager): String = {
+    optionsManager.getBuildFileName(outputSuffix, outputFileNameOverride)
+  }
+}
+
+trait HasFirrtlOptions {
+  self: ExecutionOptionsManager =>
+  var firrtlOptions = FirrtlExecutionOptions()
+
+  parser.note("firrtl options")
+
+  parser.opt[String]("input-file")
+    .abbr("i")
+    .valueName ("<firrtl-source>")
+    .foreach { x =>
+      firrtlOptions = firrtlOptions.copy(inputFileNameOverride = x)
+    }.text {
+      "use this to override the top name default, default is empty"
+    }
+
+  parser.opt[String]("output-file")
+    .abbr("o")
+    .valueName ("<output>").
+    foreach { x =>
+      firrtlOptions = firrtlOptions.copy(outputFileNameOverride = x)
+    }.text {
+      "use this to override the default name, default is empty"
+    }
+
+  parser.opt[String]("compiler")
+    .abbr("fc")
+    .valueName ("<high|low|verilog>")
+    .foreach { x =>
+      firrtlOptions = firrtlOptions.copy(compilerName = x)
+    }
+    .validate { x =>
+      if (Array("high", "low", "verilog").contains(x.toLowerCase)) parser.success
+      else parser.failure(s"$x not a legal compiler")
+    }.text {
+      s"compiler to use, default is ${firrtlOptions.compilerName}"
+    }
+
+  parser.opt[String]("info-mode")
+    .valueName ("<ignore|use|gen|append>")
+    .foreach { x =>
+      firrtlOptions = firrtlOptions.copy(infoModeName = x.toLowerCase)
+    }
+    .validate { x =>
+      if (Array("ignore", "use", "gen", "append").contains(x.toLowerCase)) parser.success
+      else parser.failure(s"$x bad value must be one of ignore|use|gen|append")
+    }
+    .text {
+      s"specifies the source info handling, default is ${firrtlOptions.infoMode}"
+    }
+
+  parser.opt[Seq[String]]("in-line")
```

Comment:
I don't think "inline" should be hyphenated. It isn't hyphenated on Wikipedia:  https://en.wikipedia.org/wiki/Inline_function